docs/guides/composing-messages.md

```diff
@@ -69,4 +69,147 @@ ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> r
+      * ChatPostEphemeralRequestBuilder
+      * ChatPostMessageRequestBuilder
+      * ChatScheduleMessageRequestBuilder
+      * ChatUpdateRequestBuilder
+
+**Adding via Gradle:**
+
+```groovy
+dependencies {
+    implementation "com.slack.api:slack-api-model-kotlin-extension:{{ site.sdkLatestVersion }}"
+    implementation "com.slack.api:slack-api-client-kotlin-extension:{{ site.sdkLatestVersion }}"
+}
+```
+
+**Adding via Gradle Kotlin DSL:**
+
+```kotlin
+dependencies {
+    implementation("com.slack.api:slack-api-model-kotlin-extension:{{ site.sdkLatestVersion }}")
+    implementation("com.slack.api:slack-api-client-kotlin-extension:{{ site.sdkLatestVersion }}")
+}
+```
+
+### Notable examples and features
+
+**Standalone withBlocks builder which comes with the model extension**:
+
+```kotlin
+import com.slack.api.model.kotlin_extension.block.withBlocks
+
+val constructedMessage = withBlocks {
+    section {
+        plainText("Now this can be passed to anything that requires a list of LayoutBlocks")
+    }
+}
+```
+
+**Type safe enums for inputs which require specific string inputs**:
+
+```kotlin
+import com.slack.api.model.kotlin_extension.block.element.ButtonStyle
+import com.slack.api.model.kotlin_extension.block.element.ConversationType
+import com.slack.api.model.kotlin_extension.block.withBlocks
+
+val message = withBlocks { 
+    section { 
+        plainText("Please select the person or group you would like to send a cat GIF to.")
+
+        // "accessory" is provided here, but it can be omitted for brevity
+        accessory { 
+            conversationsSelect { 
+                // Or alternatively, provide strings via filter("im", "mpim") if you'd prefer
+                filter(ConversationType.IM, ConversationType.MULTIPARTY_IM)
+                placeholder("Where should we send the cat?")
+                
+                confirm { 
+                    title("Confirm destination")
+                    plainText("Are you sure you want to send a cat GIF to this person or group?")
+                    confirm("Yes, send it")
+                    deny("Don't send it")
+                    
+                    style(ButtonStyle.PRIMARY)
+                }
+            }
+        }
+    }
+}
+```
+
+**Write DSL extension functions for message templating**:
+
+```kotlin
+import com.slack.api.model.kotlin_extension.block.ActionsBlockBuilder
+import com.slack.api.model.kotlin_extension.block.withBlocks
+
+fun ActionsBlockBuilder.presentOptions(vararg optionNames: String, prompt: String? = null) {
+    staticSelect { 
+        if (prompt != null) {
+            placeholder(prompt)
+        }
+        
+        options {
+            for (optionName in optionNames) {
+                option {
+                    plainText(optionName)
+                    value(optionName.toLowerCase())
+                }
+            }
+        }
+    }
+}
+
+val message = withBlocks { 
```

Comment:
This is not a message but an array - blocks.

Also, we may want to have modal examples too. We can have several sections for message, modal, and Home tab.

Reply:
Would you like me to add parallel examples using the Block Kit DSL to `docs/guides/modals.md` and `docs/guides/app-home.md` for this? I've done it in the latest set of changes, but I can put them somewhere else if need be.

Should I also add a `.blocks { }` extension function on the View builder as part of the `slack-api-model` kotlin extension package? Seems appropriate since we did it for the chat builders. I've also added that as part of this MR but I can remove it if you want.

Reply:
>I've done it in the latest set of changes, but I can put them somewhere else if need be.

Thanks a lot, those additions look perfect to me!

> Should I also add a .blocks { } extension function on the View builder as part of the slack-api-model kotlin extension package?

This is a good catch 🙇  We should have the extension function but I prefer making another PR with some simple test code. We can merge the code change whenever it becomes ready. By contrast, we need to hold off merging this document PR until the timing of v1.1.0 release.

Reply:
> making another PR with some simple test code

If you're busy for others, I can take this task - please feel free to let me know if you want me to do it.

Reply:
Sure, I'll pull the extension function into another pull request and add a test for it

Reply:
The current branch is now updated to not have the extension function

Reply:
View extension function is in #504